src/sentry/scripts/quotas/is_rate_limited.lua

```diff
@@ -24,7 +24,8 @@ local results = {}
 local failed = false
 for i=1, #KEYS, 2 do
     local limit = tonumber(ARGV[i])
-    local rejected = (redis.call('GET', KEYS[i]) or 0) - (redis.call('GET', KEYS[i + 1]) or 0) + 1 > limit
+    -- limit=-1 means "no limit"
+    local rejected = limit ~= -1 and (redis.call('GET', KEYS[i]) or 0) - (redis.call('GET', KEYS[i + 1]) or 0) + 1 > limit
```

Comment:
_nit_: Can we restructure this for readability? For example:

```suggestion
    local rejected = false
    
    -- limit = -1 indicates "no limit"
    if limit >= 0 then
        rejected = (redis.call('GET', KEYS[i]) or 0) - (redis.call('GET', KEYS[i + 1]) or 0) + 1 > limit
    end
```